L1Trigger/TrackFindingTracklet/src/MatchEngineUnit.cc

```diff
@@ -107,9 +108,21 @@ void MatchEngineUnit::step() {
 
 void MatchEngineUnit::processPipeline() {
   if (good___) {
-    bool isPSmodule = vmstub___.isPSmodule();
     int stubfinerz = vmstub___.finerz().value();
     int stubfinephi = vmstub___.finephi().value();
+    bool isPSmodule = false;
+
+    if (barrel_) {
+      isPSmodule = layerdisk_ < N_PSLAYER;
+    } else {
+      const int absz = (1 << settings_.MEBinsBits()) - 1;
+      if (layerdisk_ < N_LAYER + 2) {
+        isPSmodule = ((rzbin___ & absz) < 3) || ((rzbin___ & absz) == 3 && stubfinerz <= 3);
```

Comment:
Where does the number "3" come from?

Reply:
@aryd do you know if we can derive the 3's from any of our constants?

Reply:
ping @bryates @aryd - could we resolve this so PR can be merged/closed?

Reply:
I just pushed an updated version of MatchEngineUnit.{cc,h} which removes the use of the hardcoded numbers and derives the cut-off for separating PS and 2S modules in the disk from geometry constants.